cdap-data-fabric/src/main/java/io/cdap/cdap/store/StoreDefinition.java

```diff
@@ -34,6 +36,8 @@ public final class StoreDefinition {
 
   private static final Logger LOG = LoggerFactory.getLogger(StoreDefinition.class);
 
+  private static List<String> batchCreateStatements;
```

Comment:
Why a list here? DDLs are being created in Spanner class, so having the list here requires these extra methods and it's a bit error-prone. 
How about we add two new methods in the StructuredTableAdmin: isBatchCreateRequired() and commit(). 

So all we need to do here in StoreDefinition is to call tableAdmin.commit() at the end if tableAdmin.isBatchCreateRequired() returns true. 

Now in Spanner implementation, create calls will only store DDLs statements in a list, and upon commit, we execute all those DDLs.

Reply:
Thanks for the suggestion Masoud.

I have moved the ddl statement retention logic in Spanner storage impl.
Regarding the create calls, I think we should keep this create & defer create method different.
This was also highlighted in this review comment by Sanket https://github.com/cdapio/cdap/pull/15743#discussion_r1856628663

Create() usually gives the perception that the method itself handles the table creation and if we just persist ddl statements here, it is prone to mistakes like batchCommit() not being called later. 
If deferCreate() is used, developer will know, we should always call batchCommit().

I have added another method similar to existing updateOrCreate - the new method is named updateOrDeferCreate.
This keeps the method name & its purpose clear.

Please take a look and let me know what you think.